hibernate-core/src/main/java/org/hibernate/type/CollectionType.java

```diff
@@ -631,6 +631,14 @@ public Object replace(
 			final Object owner,
 			final Map<Object, Object> copyCache) throws HibernateException {
 		if ( original == null ) {
+			if ( target == null ) {
+				return null;
+			}
+			if ( target instanceof PersistentCollection ) {
+				final Collection collection = (Collection) target;
+				collection.clear();
+				return collection;
+			}
```

Comment:
@beikov I added a check that `target` is a `PersistentCollection` in this way calling `clear` should be safe

Reply:
That's fine, though I thought you wrote that only a `PersistentCollection` can occur at this point, so I would rather expect an assert statement. It's fine either way, just want to make sure the expectations are correct.

Reply:
I wrote that I though not that i was sure :)